esp-hal/src/timer/timg.rs

```diff
@@ -96,17 +90,16 @@ static INT_ENA_LOCK: [Lock; NUM_TIMG] = [const { Lock::new() }; NUM_TIMG];
 #[cfg_attr(not(timg_timer1), doc = "a general purpose timer")]
 #[cfg_attr(timg_timer1, doc = "2 timers")]
 /// and a watchdog timer.
-pub struct TimerGroup<'d, T, DM>
+pub struct TimerGroup<T>
 where
     T: TimerGroupInstance,
-    DM: Mode,
 {
-    _timer_group: PeripheralRef<'d, T>,
+    _timer_group: T,
     /// Timer 0
-    pub timer0: Timer<Timer0<T>, DM>,
+    pub timer0: Timer,
     /// Timer 1
     #[cfg(timg_timer1)]
-    pub timer1: Timer<Timer1<T>, DM>,
+    pub timer1: Timer,
     /// Watchdog timer
     pub wdt: Wdt<T>,
```

Comment:
Any reason not to erase the watchdog as well? Or you just didn't want to deal with it in this PR?